src/Prescription/Prescription.controller.ts

```diff
@@ -0,0 +1,204 @@
+import { PrescriptionGuard } from '../Guards/Prescription.guard';
+
+// import { FhirService } from 'src/Integerations/fhirService';
+
+@ApiTags('prescriptions')
+@Controller('prescription')
+@UseGuards(PrescriptionGuard)
+export class PrescriptionController {
+  constructor(
+    private readonly prescriptionService: PrescriptionService, // private readonly fhirService: FhirService  //fhirService to Interact with External API
+  ) {}
+
+  @Get('search')
+  @ApiQuery({
+    name: 'page',
+    required: false,
+    type: Number,
+    description: 'Page number',
+  })
+  @ApiQuery({
+    name: 'size',
+    required: false,
+    type: Number,
+    description: 'Number of items to return',
+  })
+  @ApiQuery({
+    name: 'nhi',
+    required: true,
+    type: String,
+    description: 'NHI number for which we want prescription records',
+  })
+  @ApiOperation({ summary: 'Get prescriptions by NHI' })
+  @ApiResponse({
+    status: 200,
+    description: 'The found record',
+    type: [CreatePrescriptionDto],
+  })
+  @UsePipes(ValidationPipe)
+
+  /**
+   * This controller method is responsible for fetching the prescription data for a specific patient.
+   * It receives the query parameters through GetPrescriptionDto.
+   *
+   * @param {GetPrescriptionDto} queryData - The query parameters received from the request.
+   * @returns {Object} responseObj - An object that contains a message and the prescriptions data.
+   *
+   * @throws {InternalServerErrorException} - Throws an exception if an error occurs during the operation.
+   */
+  async searchPrescriptionsForPatient(@Query() queryData: GetPrescriptionDto) {
+    try {
+      // Fetches the prescription data for the patient based on the query parameters
+      const prescriptionData =
+        await this.prescriptionService.searchPrescriptionsForPatient(queryData);
+      // Check if there are any prescriptions for the given NHI otherwise throw error
+      if (!_.isEmpty(prescriptionData)) {
+        //Sync the External refernece on backgroud, since we doesn't want to wait until the syncing is complete
+        // this.fhirService.createOrUpdateMedicationRequest(prescriptions)
+        return {
+          message: 'Success',
+          data: prescriptionData,
+        };
+      }
+      //if no record found in our database then fetch it from external API
+      // prescriptionData = await this.fhirService.getPatientPrescriptionRecords(
+      //   queryData.nhi,
+      // );
+
+      //Transform the external API reference data to our system specific data
+      // prescriptionData = this.prescriptionService.transformExternalPrescriptionData(prescriptionData);
+
+      //If we have records in prescription data then create in our database and fetch again to get
+      //correct result (our db ids)
+      // if (!_.isEmpty(prescriptionData)) {
+      //   await this.prescriptionService.createOrUpdatePrescriptions(
+      //     prescriptionData,
+      //   );
+      //   prescriptionData =
```

Comment:
I do not think we need to make another search request, as I believe we will be fetching the data from External API in a paginated manner also. So, once data get transformed it can be show directly